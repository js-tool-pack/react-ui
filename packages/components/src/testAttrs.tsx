import { fireEvent, render } from '@testing-library/react';
import type { PropsBase } from '@pkg/shared';
import React from 'react';

export function testAttrs<
  T extends { attrs?: Partial<React.HTMLAttributes<HTMLElement>> },
>(component: React.FC<T>, displayName?: string) {
  const name = displayName || component.displayName || component.name;
  test(name ? `${name} attrs` : `attrs`, () => {
    const onClick = jest.fn();

    const FC = component as React.FC<PropsBase>;
    const { container } = render(
      <FC
        attrs={{
          style: { width: '20px' },
          className: 'foo',
          role: 'note',
          onClick,
        }}
      />,
    );

    expect(container.firstChild).toHaveClass('foo');
    expect(container.firstChild).toHaveAttribute('role', 'note');
    expect(container.firstChild).toHaveStyle({ width: '20px' });

    expect(onClick).not.toBeCalled();
    fireEvent.click(container.firstChild!);
    expect(onClick).toBeCalled();
  });
}
