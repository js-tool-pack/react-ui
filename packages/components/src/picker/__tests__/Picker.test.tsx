import { PickerOption, PickerPanel, Picker } from '..';
import { render } from '@testing-library/react';
import { testAttrs } from '~/testAttrs';

describe('Picker', () => {
  testAttrs(Picker);
  testAttrs(PickerPanel);

  test('basic', () => {
    const { container } = render(
      <PickerPanel
        options={[createLetterOptions('A', 2), createLetterOptions('0', 2)]}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  test('basic2', () => {
    const { container } = render(
      <Picker
        options={[createLetterOptions('A', 2), createLetterOptions('0', 2)]}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  function createLetterOptions(char: string, length: number): PickerOption[] {
    return Array.from({ length }).map((_, i) => {
      const letter = String.fromCharCode(char.charCodeAt(0) + i);
      return {
        label: letter,
        value: letter,
      };
    });
  }
});
