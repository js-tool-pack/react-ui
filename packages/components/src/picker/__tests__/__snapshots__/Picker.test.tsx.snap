// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker basic 1`] = `
<div
  class="t-picker-panel"
>
  <ul
    class="t-picker-col"
  >
    <li
      class="t-option t-picker-col__option t--size-m"
      role="option"
    >
      <div
        class="t-option__label"
      >
        A
      </div>
    </li>
    <li
      class="t-option t-picker-col__option t--size-m"
      role="option"
    >
      <div
        class="t-option__label"
      >
        B
      </div>
    </li>
  </ul>
  <ul
    class="t-picker-col"
  >
    <li
      class="t-option t-picker-col__option t--size-m"
      role="option"
    >
      <div
        class="t-option__label"
      >
        0
      </div>
    </li>
    <li
      class="t-option t-picker-col__option t--size-m"
      role="option"
    >
      <div
        class="t-option__label"
      >
        1
      </div>
    </li>
  </ul>
</div>
`;
